### ISSUE TITLE
[MD056] false positive with `flavor = "mkdocs"` and pipe in code block

### ISSUE THREAD

**Issue by FBruzzesi** [rvben/rumdl#165]
# Description

> [!IMPORTANT]
> I hope I am not misunderstanding how to use the `flavor = "mkdocs"` flag

mkdocs renders pipe character within code without escaping it. However even when `flavor = "mkdocs"` this is flagged as an issue.

## How to reproduce

**mkdocs.yaml**
```yaml
site_name: repro

nav:
  - index.md
```

**.rumdl.toml**

```toml
[global]
flavor = "mkdocs"
```

**docs/index.md**

```md
# Example table

| Python type | Example  |
| ----------- | -------- |
| Union       | `x | y`  |
```

Then: 

```terminal
uv tool run rumdl version && uv tool run rumdl check .
rumdl 0.0.182
docs/index.md:5:1: [MD056] Table row has 3 cells, but expected 2 [*]

Issues: Found 1 issues in 1 file (9ms)
Run `rumdl fmt` to automatically fix 1 of the 1 issues
```

> [!NOTE]
> You can see I updated the version now  😂

However if we render it:

```terminal
PYTHON_GIL=0 uv tool run mkdocs serve -a localhost:8001
```

we get no issue

<img width="1261" height="295" alt="Image" src="https://github.com/user-attachments/assets/34ccb0b8-eed6-412e-a33a-9caff7ad72cf" />

**Comment by rvben** [comment]
Thanks for raising this, @FBruzzesi. There was indeed a difference between how the different flavors handled this, and that was not being accounted for.

This should be fixed in v0.0.183 (just released). 👍
